plugin/src/node-builder.ts

```diff
@@ -4,12 +4,16 @@ import { createRemoteFileNode } from "gatsby-source-filesystem";
 // 'gid://shopify/Metafield/6936247730264'
 export const pattern = /^gid:\/\/shopify\/(\w+)\/(.+)$/;
 
-function attachParentId(obj: Record<string, any>, gatsbyApi: SourceNodesArgs) {
+function attachParentId(
+  obj: Record<string, any>, 
+  gatsbyApi: SourceNodesArgs,
+  pluginOptions: ShopifyPluginOptions,
+) {
   if (obj.__parentId) {
     const [fullId, remoteType] = obj.__parentId.match(pattern) || [];
     const field = remoteType.charAt(0).toLowerCase() + remoteType.slice(1);
     const idField = `${field}Id`;
-    obj[idField] = gatsbyApi.createNodeId(fullId);
+    obj[idField] = gatsbyApi.createNodeId(`${pluginOptions.typePrefix}_${fullId}`);
```

Comment:
This is the last part I'm not sure of, I think. It could impact existing users (though there aren't many) if we start using a different string to generate their node IDs. So even though it makes the code more complicated, we probably should omit the underscore if there's no type prefix after all.

Reply:
Valid! Since `pluginOptions.typePrefix` defaults to empty string I replaced it with `gatsbyApi.createNodeId(`${pluginOptions.typePrefix}${fullId}`)` and it should behave as you ask for. Fixed!